lib/Elastica/Multi/ResultSet.php

```diff
@@ -61,19 +61,22 @@ protected function _init(Response $response, array $searches)
         if (isset($responseData['responses']) && is_array($responseData['responses'])) {
             reset($searches);
             foreach ($responseData['responses'] as $key => $responseData) {
-                $currentSearch = each($searches);
+                //todo in php 7.3 $firstKeyCurrentSearch = array_key_first($array);
+                $firstKeyCurrentSearch = key($searches);
+                reset($searches);
+                $currentSearch = array_shift($searches);
 
-                if ($currentSearch === false) {
+                if ($currentSearch === null) {
                     throw new InvalidException('No result found for search #'.$key);
-                } elseif (!$currentSearch['value'] instanceof BaseSearch) {
+                } elseif (!$currentSearch instanceof BaseSearch) {
```

Comment:
Have you considered splitting those *if-elseif* into two separate *if*'s? It may look way more readable here.